code/modules/admin/verbs/debug.dm

```diff
@@ -1540,4 +1540,38 @@ var/obj/blend_test = null
 
 	vote.forced_map = rigged_choice
 
+/client/proc/check_for_unconnected_atmos()
+	set category = "Debug"
+	set name = "Check Vent/Scrubber Connections"
+	set desc = "Outputs a list of all vents and scrubbers that aren't connected to the main station's pipe network."
+
+	if(!check_rights(R_DEBUG))
+		return
+
+	var/search_entire_world = 0
+	if(alert(usr, "Search the entire world, or just the station z-level?", "Specify scope of search", "Entire world!", "Just the station.") == "Entire world!")
+		search_entire_world = 1
+
+/* If search_entire_world is set, search all pumps and scrubbers. Otherwise, only search pumps and scrubbers on the map's designated main station z-level. */
+	var/list/unconnected_atmos = list()
+	for(var/obj/machinery/atmospherics/unary/vent_pump/V in world)
```

Comment:
All atmos machine are added to a global list, aren't they? It would be faster to iterate through that instead. If this runs instantly you may not bother, though.

Reply:
I wasn't sure if stuff was removed from the `machines` global list when depowered/not processing, so I just went with `world`. The verb was always instant for me, but I'll investigate changing it to the machines list now and see if that works.